xds/src/main/java/io/grpc/xds/LoadReportClient.java

```diff
@@ -421,5 +459,75 @@ void sendLoadStatsRequest(List<ClusterStats> clusterStatsList) {
     void sendError(Exception error) {
       lrsRequestWriterV3.onError(error);
     }
+
+    private io.envoyproxy.envoy.config.endpoint.v3.ClusterStats buildClusterStats(
+        ClusterStats stats) {
+      io.envoyproxy.envoy.config.endpoint.v3.ClusterStats.Builder builder =
+          io.envoyproxy.envoy.config.endpoint.v3.ClusterStats.newBuilder()
+              .setClusterName(stats.clusterName());
+      if (stats.clusterServiceName() != null) {
+        builder.setClusterServiceName(stats.clusterServiceName());
+      }
+      for (UpstreamLocalityStats upstreamLocalityStats : stats.upstreamLocalityStatsList()) {
+        builder.addUpstreamLocalityStats(
+            io.envoyproxy.envoy.config.endpoint.v3.UpstreamLocalityStats.newBuilder()
+                .setLocality(
+                    io.envoyproxy.envoy.config.core.v3.Locality.newBuilder()
+                        .setRegion(upstreamLocalityStats.locality().region())
+                        .setZone(upstreamLocalityStats.locality().zone())
+                        .setSubZone(upstreamLocalityStats.locality().subZone()))
+            .setTotalSuccessfulRequests(upstreamLocalityStats.totalSuccessfulRequests())
+            .setTotalErrorRequests(upstreamLocalityStats.totalErrorRequests())
+            .setTotalRequestsInProgress(upstreamLocalityStats.totalRequestsInProgress())
+            .setTotalIssuedRequests(upstreamLocalityStats.totalIssuedRequests()));
+      }
+      for (DroppedRequests droppedRequests : stats.droppedRequestsList()) {
+        builder.addDroppedRequests(
+            io.envoyproxy.envoy.config.endpoint.v3.ClusterStats.DroppedRequests.newBuilder()
+                .setCategory(droppedRequests.category())
+                .setDroppedCount(droppedRequests.droppedCount()));
+      }
+      return builder
+          .setTotalDroppedRequests(stats.totalDroppedRequests())
+          .setLoadReportInterval(Durations.fromNanos(stats.loadReportIntervalNano()))
+          .build();
+    }
+  }
+
+  /**
+   * Converts Java representation of the given JSON value to protobuf's {@link
+   * com.google.protobuf.Value} representation.
+   *
+   * <p>The given {@code rawObject} must be a valid JSON value in Java representation, which is
+   * either a {@code Map<String, ?>}, {@code List<?>}, {@code String}, {@code Double}, {@code
+   * Boolean}, or {@code null}.
+   */
+  private static Value convertJsonValueToProtobufValue(Object rawObject) {
```

Comment:
Where is it used?

Reply:
Oh, initially I was trying to migrate `Node` as well. But then found Node is much more complicated. Deleted this method.